### PR TITLE
Components: Use real timers where possible

### PR DESCRIPTION
**Pull request by tyxla**
## What?
This PR updates component tests to use real timers where possible. This is a follow-up to #46714.

## Why?
We discovered that Jest real timers are preferable when their use is possible, see https://github.com/facebook/react/issues/25889 and the description in #46714.

## How?
We're removing the specific `jest.useFakeTimers();` calls, and the unnecessary `advanceTimers: jest.advanceTimersByTime`  in the user-event setup. We're also removing the now unnecessary `jest.advanceTimersByTime()` calls and post-test cleanup that we've been doing with `jest.runOnlyPendingTimers()` and `jest.useRealTimers()`.

## Testing Instructions
Verify all tests still pass.

### Testing Instructions for Keyboard
None

## Screenshots or screencast 
None.

### CODE REVIEW

**Comment by Mamaduka** [comment]
> We discovered that Jest real timers are preferable when their use is possible, see https://github.com/facebook/react/issues/25889 and the description in https://github.com/WordPress/gutenberg/pull/46714.

I think we only prefer real times because of the bug @jsnajdr discovered.

How does this affect JS Unit test runtime? But the difference probably won't be too big.

File: packages/components/src/color-picker/test/index.js [hunk 1]
```diff
@@ -43,11 +41,7 @@ function moveReactColorfulSlider( sliderElement, from, to ) {
 	fireEvent( sliderElement, new FakeMouseEvent( 'mousemove', to ) );
 }
 
-const sleep = ( ms ) => {
-	const promise = new Promise( ( resolve ) => setTimeout( resolve, ms ) );
-	jest.advanceTimersByTime( ms + 1 );
-	return promise;
-};
+const sleep = ( ms ) => new Promise( ( resolve ) => setTimeout( resolve, ms ) );
```
Comment (jsnajdr) [inline packages/components/src/color-picker/test/index.js]: This should probably get replaced with something like:
```js
await waitFor( () => expect( onChange ).toHaveBeenCalled() );
```
at all the places where `sleep()` is used.

**Review by jsnajdr — APPROVED** [review]

**Comment by jsnajdr** [comment]
> I think we only prefer real times because of the bug @jsnajdr discovered.

There are also other, semi-philosophical but IMO very good reasons to prefer real timers when working with React Testing Library. Here I'm posting publicly what was originally an internal P2 comment in a discussion about Tumblr, but it applies universally: https://jsnajdr.wordpress.com/2023/01/11/prefer-jest-real-timers-when-testing-with-react-testing-library/

**Comment by Mamaduka** [comment]
Thanks for sharing, @jsnajdr 🙇

File: packages/components/src/color-picker/test/index.js (hunk 1 above)
Comment (tyxla) [reply packages/components/src/color-picker/test/index.js]: Thanks for the suggestion, done in https://github.com/WordPress/gutenberg/pull/47056/commits/8e64eb4489124c7d5148b45afbab648b8c346b38

Comment (jsnajdr) [reply packages/components/src/color-picker/test/index.js]: There are still the "we need to sleep for at least 1ms" comments that no longer make any sense.

Comment (tyxla) [reply packages/components/src/color-picker/test/index.js]: I mean, they make sense IMHO but I'm happy to remove them.

Comment (tyxla) [reply packages/components/src/color-picker/test/index.js]: Removed in https://github.com/WordPress/gutenberg/pull/47056/commits/5b2ae5aaff39f98c60596f3969ecaadccfa92c5f

Comment (jsnajdr) [reply packages/components/src/color-picker/test/index.js]: You're right, now I realize the actually do make sense. I apologize for being so blunt.

Comment (tyxla) [reply packages/components/src/color-picker/test/index.js]: No worries. I think you have a point that the `waitFor()` calls should be self-explanatory, making the comment redundant.

**Comment by ciampo** [comment]
Thank you all for working on this, and special thanks to @jsnajdr for sharing his detailed post re. fake vs real timers 👏

**Pull request merged**